### ISSUE TITLE
Can not make readers public if also providing default values

### ISSUE THREAD

**Issue by derosm2** [pawurb/smart_init#3]
Looks like if a default value is provided, then the public_readers option is ignored:
```rb
class PublicApiClient < SmartInit::Base
  initialize_with :network_provider, api_token: "ABC", public_readers: true
end

client = PublicApiClient.new(network_provider: Faraday.new, api_token: 'secret_token')
client.network_provider
NoMethodError: private method `network_provider'
```

**Comment by pawurb** [comment]
@derosm2 thanks for spotting. Fixed and released at 4.1.0

**Comment by derosm2** [comment]
Thanks for the quick turnaround!
